### DISCUSSION TITLE
GenICam Library

### DISCUSSION THREAD

**Discussion by FaisalFagihi** [Stira-sa/GigeVision#2] [Show and tell]
- GenICam is the right way to implement GigeVision Protocol.

- [GenICam_Standard_v2_1_1.pdf](https://github.com/Stira-sa/GigeVision/files/7347185/GenICam_Standard_v2_1_1.pdf) documentation been followed while implementing the library. 

- Ready to use and been tested on 5 models:
  - SBGigE
  - iPORTCLGigE
  - CXG_IP
  - iPORTAnalogProOEM
  - GEV_B1020C

- There are some nodes have not been implemented yet: 
  - DocuURL
  - EventID
  - ExposeStatic
  - Extension
  - ImposedAccessMode
  - IsDepecated
  - MergePriority
  - pAlias
  - pBlockPolling
  - pCastAlias
  - pError
  - pInvalidator
  - pIsAvailable
  - pIsImplemented
  - pIsLocked
  - pSelected

Please feel free to submit issues or comments to improve the library.

**Comment by FaisalFagihi** [comment]
@Touseefelahi

**Comment by Touseefelahi** [comment]
As we are aware that there are some of the features that are not working properly in the XML parsing part. 
For our application needs, there is no freedom of having this critical task having any sort of problem. We should implement the parser 100% according to the GenICam standard so that it can be used with any type of camera.

To solve the described issue, even if we have to develop the parser from scratch we can schedule it accordingly.

@FaisalFagihi Let's analyze the problem and see if it is feasible to build it from scratch, or we can modify the existing code to fulfill the need.
related issue here #14
